#! /usr/bin/env python

PACKAGE = 'lane_departure_checker'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator ()

gen.add("footprint_margin", double_t, 0, "footprint_margin", 0.0, 0.0, 1.0)
gen.add("resample_interval", double_t, 0, "resample_interval", 0.3, 0.01, 3.0)
gen.add("max_deceleartion", double_t, 0, "max_deceleartion", 3.0, 0.1, 5.0)
gen.add("delay_time", double_t, 0, "delay_time", 0.3, 0.0, 3.0)
gen.add("max_lateral_deviation", double_t, 0, "max_lateral_deviation", 1.0, 0.0, 5.0)
gen.add("max_longitudinal_deviation", double_t, 0, "max_longitudinal_deviation", 1.0, 0.0, 5.0)
gen.add("max_yaw_deviation_deg", double_t, 0, "max_yaw_deviation_deg", 30.0, 0.0, 90.0)

exit (gen.generate (PACKAGE, "lane_departure_checker", "LaneDepartureChecker"))
